target/linux/realtek/files-6.12/drivers/net/dsa/rtl83xx/common.c

```diff
@@ -382,6 +382,14 @@ static int __init rtl83xx_mdio_probe(struct rtl838x_switch_priv *priv)
 				return -ENODEV;
 			}
 		}
+		
+		if (pcs_node) {
+			priv->ports[pn].swap_rx_polarity = of_property_read_bool(pcs_node, "swap_rx_polarity");
+			priv->ports[pn].swap_tx_polarity = of_property_read_bool(pcs_node, "swap_tx_polarity");
```

Comment:
The new properties must be prefixed to make them vendor specific. It seems like `realtek,` is the current consensus for the prefix.

The properties should also use dashes and not underscores. And I've often seen names like:

```
realtek,pnswap-tx
realtek,pnswap-rx
```

for these things.

And this brings me to the next thing. The PHY driver in OpenWrt already supports this on the PHY side. We therefore need a really good explanation why we need now a patch on the SerDes/SoC side.

See ddb0cd276cff667267c10e13d5ad0506e9d4ec22 and #20227 in OpenWrt

Reply:
if mediatek-filogic requires the driver of `RTL8261N` to support the swap itself instead of being able to use the mediatek,pnswap-rx/airoha,pnswap-rx like attributes in the dts which I assume are on the soc(?), then I'm not sure whether these can be removed from the driver and moved to serdes unless Mediatek can also access this via serdes (how it's apparently supposed to be)

What do you think? These are just assumptions based on what I just read and could be wrong

Reply:
That code doesn't get compiled if you build for realtek. Check the ifdefs in the code.